### DISCUSSION TITLE
Issue form preview in local (such as on Visual Studio Code)

### DISCUSSION THREAD

**Discussion by noritada** [community/community#7039] [Projects and Issues]
I prefer a workflow with a local editor, and the migration from Markdown issue templates to YAML issue forms was also done in local Visual Studio Code. I was a little disappointed that I couldn't preview the YAML files until I committed, pushed, and opened the YAML files in a branch on GitHub. The integration between Visual Studio Code and GitHub is quite good, so it would be very helpful to be able to preview and check locally.

I am very happy that Visual Studio Code correctly recognized this special file and auto-completed the input appropriately. However, when adding a new component to the body, the word is sometimes not recognized. In that case, deleting the text and retrying inputting worked. I will report this just in case.

Finally, I am very happy that you are providing Issue forms. I've seen a lot of Markdown issue templates (with HTML comments, etc.) and thought it would be nice to have a better way, so I really like this form. I hope it will be available in the GHEC internal repositories soon.

**Comment by Wladefant** [comment]
Yes that would be great. It seems like the only option is the [YAML Preview](https://marketplace.visualstudio.com/items?itemName=daiwata.yaml-preview) extension that tries to put everything in a table (not great). The YAML extension form RED hat and the prettier extension seems to be also great for the syntax.

**Comment by mloskot** [comment]
I'm also interested in local preview of GitHub issue forms.

The only _workaround_ that allows to preview the issue form before it is committed to the main branch of repository is this, not local though:

1. Submit GitHub issue form changes via pull request
2. Go to PR > Files changed > meatballs menu on the right of your form YAML file > View file
3. The issue form is rendered, but read-only. 

I posted some screenshots to the [Make issue forms available on private repositories](https://github.com/orgs/community/discussions/4264#discussioncomment-5311827) thread.

**Reply by nbro** [reply]
Btw, in case you get errors while using this approach, you can check this: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/common-validation-errors-when-creating-issue-forms#bodyi-label-must-be-a-string

**Reply by tyler-dane** [reply]
Although this doesn't help with previewing the form, the [`YAML` VSCode extension](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) from RedHat includes lint rules for GitHub's issue forms, which could cut down on the # of times you'd need to preview the file.

**Reply by reespozzi** [reply]
cc @mloskot @tyler-dane https://github.com/orgs/community/discussions/7039#discussioncomment-15174732

**Comment by reespozzi** [comment]
I built this, which may be of interest here when I had a similar issue; https://marketplace.visualstudio.com/items?itemName=ReesPozzi.github-issue-template-preview&ssr=false#overview

I can now render the GitHub issue template in VS Code
